File: akka-essentials/Chapter5/AkkaDispatcherExample/src/main/scala/org/akka/essentials/dispatcher/example/Dispatcher/Example2.scala
package org.akka.essentials.dispatcher.example.Dispatcher

import akka.actor.ActorSystem
import com.typesafe.config.ConfigFactory
import akka.actor.Props
import akka.routing.RoundRobinRouter
import org.akka.essentials.dispatcher.MsgEchoActor

object Example2 {
    def main(args: Array[String]): Unit = {}

    val _system = ActorSystem.create("default-dispatcher", ConfigFactory.load().getConfig("MyDispatcherExample"))

    val actor = _system.actorOf(Props[MsgEchoActor].withDispatcher("defaultDispatcher1").withRouter(
        RoundRobinRouter(5)))

    0 to 25 foreach {
        i => actor ! i
    }
    _system.shutdown()
}

File: akka-essentials/Chapter5/AkkaDispatcherExample/src/main/java/org/akka/essentials/dispatcher/java/example/Dispatcher/Example2.java
package org.akka.essentials.dispatcher.java.example.Dispatcher;

import akka.routing.RoundRobinPool;
import org.akka.essentials.dispatcher.java.MsgEchoActor;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

import com.typesafe.config.ConfigFactory;

public class Example2 {

    /**
     * @param args
     */
    public static void main(String[] args) {
        ActorSystem _system = ActorSystem.create("default-dispatcher",
                ConfigFactory.load().getConfig("MyDispatcherExample"));

        ActorRef actor = _system.actorOf(Props.create(MsgEchoActor.class)
                .withDispatcher("defaultDispatcher1").withRouter(
                        new RoundRobinPool(5)));

        for (int i = 0; i < 25; i++) {
            actor.tell(i, ActorRef.noSender());
        }

        _system.shutdown();

    }

}
